### ISSUE TITLE
Replace string equality comparison on tests with a weaker string comparison

### ISSUE THREAD

**Issue by henvic** [launchpad-project/cli#8]
Currently == is being used to verify string comparison. Writing tests like this is getting harder and harder. Besides issues with supporting multiple platforms that may arise due to carrier return and line feed, a leading whitespace, or extra line at end of processes... A nightmare.

This kind of comparison should ignore spacing characters. But preferably normalize new lines so line breaking context is not lost instead of stripping them entirely.

**Comment by cirocosta** [comment]
Hey @henvic, couldn't you simply use some sort of asserting library for that? I'm not aware of go's ecosystem, are there many tiny modules that does a single thing like we have in js?

**Comment by henvic** [comment]
@cirocosta,
tl;dr: relying on many tiny modules to do small things is definitely not Go's ecosystem way to do things. Peculiar to Go ecosystem is the predominance of if conditionals, rather than assertions (* [Why does Go not have assertions?](https://golang.org/doc/faq#assertions)).

That being said, I actually use one or two assertions (such as for comparing JSON to a map) and this is a case where an assertion might actually simplify things (I don't really know if I am going to use an external library for that, but I guess not).

* This is why you can find odd-looking tests with if conditionals where assertions would be expected for anyone familiar with other languages https://github.com/launchpad-project/cli/blob/5e57aa2/containers/containers_test.go#L67-L75

That being said, I use a few assertions for very specific things such as for validating JSON and this is such a case imho.

**Comment by cirocosta** [comment]
hmmm, interesting. Their arguments makes sense :+1:

**Comment by henvic** [comment]
Closing with 75a97bd23316426d99dbc44313b10590ee27c7f6.

**Comment by henvic** [comment]
@cirocosta,

[LeftPad and Go: can tooling help?](https://divan.github.io/posts/leftpad_and_go/)

> A little copying is better than a little dependency.
~ [Go Proverbs](http://go-proverbs.github.io/)

**Comment by cirocosta** [comment]
Awesome. Thanks for the reference @henvic . Also, i'll definitely checkout the talk pointed in the article (Go proverbs - Rob Pike @ Gopherphest)
